python/ray/data/datasource/file_based_datasource.py

```diff
@@ -217,6 +220,11 @@ def expand_paths(
     ) -> Tuple[List[str], List[Optional[int]]]:
         from pyarrow.fs import FileType
 
+        logger.warning(
```

Comment:
I'm not sure we should log this unconditionally: the input paths may contain directory so it's not appropriate to suggest FastFileMetadataProvider. How about commenting the class that if users are dealing with only files, they should use FastFileMetadataProvider?